apps/full-stack-tests/src/hierarchies/HierarchyFiltering.test.ts

```diff
@@ -1528,19 +1531,24 @@ describe("Hierarchies", () => {
                 parentKeys: [...parentNode.parentKeys, parentNode.key],
                 children: false,
               };
-              const filteringProps = extractFilteringProps([], parentNode);
+              const filteringProps = extractFilteringProps(filteringPaths, parentNode);
               if (!filteringProps) {
                 return createAsyncIterator([myNode]);
               }
               const nodeMatchesFilter =
                 filteringProps.hasFilterTargetAncestor ||
-                filteringProps.filteredNodePaths?.some((fp) => {
-                  const { path } = HierarchyFilteringPath.normalize(fp);
+                filteringProps.filterPathsIdentifierPositions?.some(([pathIndex, identifierIndex]) => {
+                  if (filteringProps.nodeIdentifierPaths.length <= pathIndex) {
+                    return false;
+                  }
+                  const nodePath = filteringProps.nodeIdentifierPaths[pathIndex];
+                  const { path } = HierarchyFilteringPath.normalize(nodePath);
+                  const identifierIndexAdded = identifierIndex + 1;
                   return (
-                    path.length &&
-                    HierarchyNodeIdentifier.isGenericNodeIdentifier(path[0]) &&
-                    path[0].source === "custom-provider" &&
-                    path[0].id === myNode.key.id
+                    path.length > identifierIndexAdded &&
+                    HierarchyNodeIdentifier.isGenericNodeIdentifier(path[identifierIndexAdded]) &&
+                    path[identifierIndexAdded].source === "custom-provider" &&
+                    path[identifierIndexAdded].id === myNode.key.id
```

Comment:
Looks like implementing hierarchy filtering in a custom hierarchy provider became much more complicated. We should not enforce any specific implementation requirements for custom providers - they should be able to choose the most convenient way to filter the hierarchy.